src/api/posts.ts

```diff
@@ -0,0 +1,25 @@
+import { Post } from '../react-app-env';
+
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getPosts(userId: string): Promise<Post[]> {
+  const response = await fetch(`${BASE_URL}/posts?userId=${userId}`);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+}
+
+export async function getPostbyId(postId:number) {
+  const response = await fetch(`${BASE_URL}/posts/${postId}`);
+
+  return response.json();
+}
```

Comment:
Here you need to change the function name. Better to use CamelCase names.
getPostById will be correct.

https://www.syncfusion.com/blogs/post/top-javascript-naming-convention.aspx - very useful article on this